### PR TITLE
LPS-146597 Apostrophe in segment property value breaks filter

### PR DESCRIPTION
**Pull request by YurenaC**
**Motivation**
----------------

Bug fixing: When a user creates a new segment with a filter that contains an apostrophe (e.g. lastName contains O'Neal), then it cannot be edit and an exception is thrown, leaving the portal broken.

**Proposed Solution**
----------------

Modify the current implementation to tolerate apostrophes in the segments filters, escaping them before reach the backend, where the ODataParser breaks as the apostrophes are expected in pairs

**Steps to Verify:**
----------------

1. Create User "Shaquille O'Neal"
2. Create new Segment
3. Add last name property with "O'Neal"
4. Save

**Expected**
Once it's saved the segment can be edited again or read

**Actual**
An error is shown when the segment is opened

### CODE REVIEW

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:stop

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/odata.es.js [hunk 1]
```diff
@@ -59,7 +87,7 @@ function valueParser(value, type) {
 		case PROPERTY_TYPES.COLLECTION:
 		case PROPERTY_TYPES.STRING:
 		default:
-			parsedValue = `'${value}'`;
+			parsedValue = escapeSingleQuotes(value);
```
Comment (claraizquierdo) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/odata.es.js:90]: `escapeSingleQuotes` doesn't always return a string with quotes, could it be a problem? Before this change parsed value was always a string with quotes.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/StringInput.es.js [hunk 2]
```diff
@@ -46,7 +65,7 @@ class StringInput extends React.Component {
 				disabled={disabled}
 				onChange={this._handleChange}
 				type="text"
-				value={value}
+				value={this.state.value}
```
Comment (claraizquierdo) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/StringInput.es.js:68]: This is just two questions: Why is it better to have the unescaped value stored in the state? I see that you update this state in the onChange, was the prop not updating its value correctly before?

**Review by claraizquierdo — COMMENTED** [review]
I just had time to review the code (not the functionality) and I left 2 comments. I will update my review when I finish reviewing the functionality.

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Review by claraizquierdo — APPROVED** [review]

**Comment by YurenaC** [comment]
ci:forward

**Pull request closed without merging**